src/common.h

```diff
@@ -17,17 +17,23 @@
 #ifndef __IOTIVITY_NODE_FUNCTIONS_INTERNAL_H__
 #define __IOTIVITY_NODE_FUNCTIONS_INTERNAL_H__
 
-#include <string>
-#include <v8.h>
 #include <nan.h>
+#include <string>
+extern "C" {
+#include <ocstack.h>
+}
 
-#define VALIDATE_CALLBACK_RETURN_VALUE_TYPE(value, typecheck, message) \
-  if (!value->typecheck()) {                                           \
-    Nan::ThrowTypeError(                                               \
-        (std::string(message) +                                        \
-         " callback return value type must satisfy " #typecheck "()")  \
-            .c_str());                                                 \
-  }
+#define TRY_CALL(callback, context, argumentCount, arguments, exceptionReturn) \
```

Comment:
I notice you're use TRY_CALL in tandem with an assignment operator.  See below.  I do not think this is valid C++.  

```c++
Local<Value> returnValue = TRY_CALL(...);
```

Enclosing a block in parentheses like you're doing on line 27 doesn't make an anonymous function in C++.  You could use lambdas and have a different syntax. I'd suggest avoiding C++ lambdas since the maintenance cost is high due to lack of familiarity and tools compatibility.  Instead, put a reference to the return value in the macro definition, and change the usage syntax to something like this.

```c++
Local<Value> returnValue;
TRY_CALL(..., returnValue);
```